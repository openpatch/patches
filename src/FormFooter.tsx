import { ReactNode } from "react";
import { Button } from "./Button";
import { ButtonGroup } from "./ButtonGroup";
import { ButtonPrimary } from "./ButtonPrimary";
import { Box } from "./Box";

export type FormFooterProps = {
  children?: ReactNode;
  disabled?: boolean;
  onSave?: () => void;
  onCancel?: () => void;
  locale?: {
    save: string;
    cancel: string;
  };
};

export const FormFooter = ({
  locale = { save: "Save", cancel: "Cancel" },
  disabled = false,
  onSave = () => {},
  onCancel = () => {},
}: FormFooterProps) => {
  function handleSave() {
    if (!disabled) {
      onSave();
    }
  }
  return (
    <Box mt="large" display="flex" justifyContent="flex-end">
      <ButtonGroup space="standard">
        <Button onClick={onCancel}>{locale.cancel}</Button>
        <ButtonPrimary onClick={handleSave} type="submit" disabled={disabled}>
          {locale.save}
        </ButtonPrimary>
      </ButtonGroup>
    </Box>
  );
};
