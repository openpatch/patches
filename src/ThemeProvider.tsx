import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
