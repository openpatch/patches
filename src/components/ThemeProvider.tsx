import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
