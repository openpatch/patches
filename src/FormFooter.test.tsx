
import { render } from "@testing-library/react";
import { FormFooter, FormFooterProps } from "./FormFooter";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FormFooterProps) =>
render(
    <ThemeProvider>
        <FormFooter {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});