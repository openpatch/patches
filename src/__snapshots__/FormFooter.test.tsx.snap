// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  margin-top: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2>*:not(style)~*:not(style) {
  margin-left: 16px;
}

.emotion-3 {
  position: relative;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 8px;
  font-weight: 600;
  -webkit-transition: all 0.1s ease-out;
  transition: all 0.1s ease-out;
  font-size: 14px;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-3:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.emotion-3:hover:enabled {
  opacity: 0.8;
}

.emotion-3:active:enabled {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(66, 153, 225, 0.5);
}

.emotion-3 span.icon-left,
.emotion-3 span.icon-right {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 span.icon-left,
.emotion-3 span.icon-right {
  width: 14px;
}

.emotion-3 span.icon-left {
  margin-right: 8px;
}

.emotion-3 span.icon-right {
  margin-left: 8px;
}

.emotion-5 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 8px;
  font-weight: 600;
  -webkit-transition: all 0.1s ease-out;
  transition: all 0.1s ease-out;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(66, 153, 225, 0.5);
}

.emotion-5 span.icon-left,
.emotion-5 span.icon-right {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-8 {
  position: relative;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 8px;
  font-weight: 600;
  -webkit-transition: all 0.1s ease-out;
  transition: all 0.1s ease-out;
  font-size: 14px;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: #0C6B58;
  color: #EFFCF6;
}

.emotion-8:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.emotion-8:hover:enabled {
  opacity: 0.8;
}

.emotion-8:active:enabled {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.emotion-8:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(66, 153, 225, 0.5);
}

.emotion-8 span.icon-left,
.emotion-8 span.icon-right {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-8 span.icon-left,
.emotion-8 span.icon-right {
  width: 14px;
}

.emotion-8 span.icon-left {
  margin-right: 8px;
}

.emotion-8 span.icon-right {
  margin-left: 8px;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0 emotion-1"
      display="flex"
    >
      <div
        class="emotion-2"
        role="group"
      >
        <button
          class="emotion-3 emotion-4"
        >
          <div
            class="emotion-5 emotion-1"
          >
            Cancel
          </div>
        </button>
        <button
          class="emotion-7 emotion-8 emotion-4"
          type="submit"
        >
          <div
            class="emotion-5 emotion-1"
          >
            Save
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
